Tests/Unit/LoRaTools/MacCommandTests.cs

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.Tests.Unit.LoRaTools
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using global::LoRaTools;
+    using global::LoRaTools.Mac;
+    using global::LoRaTools.Regions;
+    using LoRaWan.Tests.Common;
+    using Newtonsoft.Json;
+    using Xunit;
+
+    public abstract class MacCommandTests
```

Comment:
What about making this generic and then constraining the type argument to `MacCommand`, as in:

```c#
public class MacCommandTests<T> where T : MacCommand`
{
    /* ... */
}
```

I think it will simplify a number of things like making `Subject` a `T` and removing the need for a type parameter on `DeserializationTest` and `FromBytesTest`.

Reply:
Addressed this in 973733e27f11acbf4ac0e2e12ed5326e4f435f9c